ios/RNWatson.swift

```diff
@@ -1,62 +1,60 @@
 //
 //  Created by Patrick cremin on 8/2/17.
 //
-
-import Foundation
 import SpeechToText
 import AVFoundation
-import RestKit
-
+import IBMSwiftSDKCore
 
 // SpeechToText
 @objc(RNSpeechToText)
 class RNSpeechToText: RCTEventEmitter {
 
   var accumulator = SpeechRecognitionResultsAccumulator()
-  var speechToText: SpeechToText?
-  var audioPlayer = AVAudioPlayer()
-  var callback: RCTResponseSenderBlock?
+  var speechToTextSession: SpeechToTextSession?
   var hasListeners = false
   var isListening = false;
 
   static let sharedInstance = RNSpeechToText()
 
-  private override init() {}
-
   override func supportedEvents() -> [String]! {
     return ["StreamingText"]
   }
 
-  @objc func initialize(_ apiKey: String) -> Void {
+  @objc func startStreaming(_ accessToken: String, config: [String: Any], errorCallback: @escaping RCTResponseSenderBlock) {
+
     do {
       let audioSession = AVAudioSession.sharedInstance();
 
       try audioSession.setCategory(AVAudioSession.Category.playAndRecord, mode: .default, options: [.defaultToSpeaker, .mixWithOthers, .allowBluetooth])
+      try audioSession.setActive(true)
     } catch {
       print(error)
     }
 
-    speechToText = SpeechToText(apiKey: apiKey)
-    speechToText?.defaultHeaders = ["X-Watson-Learning-Opt-Out": "true"]
-  }
-
-  @objc func startStreaming(_ config: [String: Any], errorCallback: @escaping RCTResponseSenderBlock) {
     self.isListening = true
     self.accumulator = SpeechRecognitionResultsAccumulator()
-    
-    var settings = RecognitionSettings(contentType: "audio/ogg")
-    settings.interimResults = true
-    settings.smartFormatting = true
+
     let languageCustomizationID = config["languageCustomizationId"] as? String
     let acousticCustomizationID = config["acousticCustomizationId"] as? String
 
-    var callback = RecognizeCallback()
+    let authenticator = BearerTokenAuthenticator(bearerToken: accessToken)
+
+    speechToTextSession = SpeechToTextSession(
+      authenticator: authenticator,
+      languageCustomizationID: languageCustomizationID,
+      acousticCustomizationID: acousticCustomizationID,
+      learningOptOut: true
+    )
```

Comment:
I would have set all these things in the initialize method.

Reply:
I know you've "cancelled" this comment but I'll answer anyway @gabrielperales, because I almost did that.

The reason I didn't was to "unify" it more with the web SDK, as things stand with this PR it's practically the same implementation for native & web, whereas before we had to do a separate `initialize` call for iOS.

We could have kept this, but then we'd need to be passing in the customization IDs into both methods, which seemed a bit messy to me 😇